### PR TITLE
tests: drivers: imu: Add unit tests for adis drivers

### PR DESCRIPTION
**Pull request by rbolboac**
Add unit tests for adis drivers using ceedling yml file. 
adis_generic_tests.c contain all tests in order to test adis.c file while test_adis16505 and test_adis1657x contain the specific tests for each device.

### CODE REVIEW

File: tests/drivers/imu/test/support/adis_generic_tests.c [hunk 1]
```diff
@@ -0,0 +1,4064 @@
+/***************************************************************************//**
```
Comment (buha) [inline tests/drivers/imu/test/support/adis_generic_tests.c:1]: why not name this file test_adis.c ? this way naming would be consistent with the other test files

**Review by CiprianRegus — APPROVED** [review]

**Review by amiclaus — APPROVED** [review]

**Comment by rbolboac** [comment]
implemented review changes:
- renamed adis_generic_tests.c to adis_tests.c 
- added readme file for running unit tests with ceedling

**Comment by rbolboac** [comment]
implemented review changes:
- added readme file for running unit tests with ceedling

File: tests/drivers/imu/test/support/adis_generic_tests.c (hunk 1 above)
Comment (rbolboac) [reply tests/drivers/imu/test/support/adis_generic_tests.c]: you are right, thank you!

**Review by amiclaus — APPROVED** [review]
lgtm

**Pull request merged**